### DISCUSSION TITLE
Run nodes in a pipeline in parallel

### DISCUSSION THREAD

**Discussion by rjoomen** [tesseract-robotics/tesseract_planning#439] [Q&A] [answered]
Is it possible to run nodes in a pipeline in parallel? It seems the SyncTask is created for exactly this pupose, but I have not managed to get anything working. For reference, this is how far I got, and I even had to disable [this check](https://github.com/tesseract-robotics/tesseract_planning/blob/master/tesseract_task_composer/core/src/task_composer_pipeline.cpp#L156):

![image](https://github.com/tesseract-robotics/tesseract_planning/assets/1153434/bb428656-82bf-4ff1-b310-f54e751b846e)

**Comment by Levi-Armstrong** [comment, marked as answer]
A pipeline should only ever have a single path through the graph. You want to use the Graph object which does support running things in parallel and some examples of this are the raster objects. I also recommend taking a look at Taskflows documentation for the scheduler to understand how it works so when setting up the graph you get the behavior you desire.

**Reply by rjoomen** [reply]
OK, clear, thanks.

**Comment by Levi-Armstrong** [comment]
[This](https://taskflow.github.io/taskflow/ConditionalTasking.html#TaskSchedulingPolicy) is a good thing to review.
